### ISSUE TITLE
Home Assistant reports Room assistant BLE Device Tracker unavailable for roughly 30 seconds

### ISSUE THREAD

**Issue by bleitchhk** [mKeRix/room-assistant#888]
I am not sure this is a bug, but seemed best place to ask.

Set up
===== 
6 Room assistance instances all on Pi0W
have set to Quorum to 4, but have not set any other cluster parameters
Speaking to Home assistant via MQTT / Home Assistant integration
Same config on all Pi0w/instances
tracking 6 BLE tags using BLE integration
set timeout on BLE to 120, no other BLE options set

Operation
======== 
All Pi0ws register with Home assistant
Cluster leader, # of instances, MQTT messaging,  all working and good in HA
correctly reports arriving home events and being away events

Issue
=====
RA sends MQTT status about every 3:40 - "home" normally"
HA hold status of "home" for about 3 minutes
HA shows status of "unavailable" for roughly 40 seconds
HA show home again on receipt of next MQTT message from cluster leader

Question
======= 
how can I make home assistant not set these to "unavailable" ?

Thanks in advance 

Brendan

**Comment by mKeRix** [comment]
"unavailable" usually pops up if the related instance lost connectivity - can you check your MQTT broker for the availability topics when your entity is "unavailable"? You can find the configured availability topics in the entity config under `homeassistant/...`, the relevant topics themselves live under `room-assistant/...`. For these distributed entities, it should be a topic for the entity itself and one for the current leader instance. If you can gather the messages sent to both the config and the availability topics around the time of your issue that should give use some insight as to why this issue is happening.

**Comment by rikdc** [comment]
I am having a similar issue, but it's inverted. The BLE device (using the iOS app) is detected but it's then marked as away after 30ish seconds, and never comes back. Restarting room-assistant on the Pi-Zeros repeats this consistently.

I have 4x Pi-Zeros, and it looks like one in the cluster is acting up:

```
Sep  3 17:06:23 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:06:23 - info - ClusterService: Removed 192.168.30.9:6425 from the cluster with id raspberrypi
Sep  3 17:06:41 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:06:41 - info - ClusterService: Added 192.168.30.9:6425 to the cluster with id raspberrypi
Sep  3 17:06:41 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:06:41 - info - ClusterService: raspberrypi has been elected as leader
Sep  3 17:06:58 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:06:58 - info - ClusterService: unit1backpi has been elected as leader
Sep  3 17:07:15 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:15 - info - BluetoothLowEnergyService: Discovered new BLE peripheral 671211950385 with ID 671211950385 and RSSI -65
Sep  3 17:07:22 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:22 - info - HomeAssistantService: Refreshing entity states
Sep  3 17:07:22 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:22 - info - ClusterService: unit1frontroom has been elected as leader
Sep  3 17:07:22 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:22 - info - ClusterService: raspberrypi has been elected as leader
Sep  3 17:07:23 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:23 - info - ClusterService: Removed 192.168.30.12:6425 from the cluster with id unit1backpi
Sep  3 17:07:36 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:36 - info - ClusterService: Added 192.168.30.12:6425 to the cluster with id unit1backpi
Sep  3 17:07:36 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:36 - info - ClusterService: unit1backpi has been elected as leader
Sep  3 17:07:38 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:38 - info - ClusterService: unit1backpi has been elected as leader
Sep  3 17:07:38 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:07:38 - info - ClusterService: unit1backpi has been elected as leader
Sep  3 17:08:02 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:08:02 - info - ClusterService: raspberrypi has been elected as leader
Sep  3 17:08:13 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:08:13 - info - ClusterService: unit1backpi has been elected as leader
Sep  3 17:08:43 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:08:43 - info - ClusterService: raspberrypi has been elected as leader
Sep  3 17:08:51 unit1FrontRoom room-assistant[17557]: 03/09/2021, 17:08:51 - info - ClusterService: unit1backpi has been elected as leader
```

**Comment by rikdc** [comment]
Update from the above: to address this I re-ran the following command on the Pi-Zeros:

` sudo apt-get install bluetooth bluez blueman` and the problem appears resolved.

**Comment by bleitchhk** [comment]
Hi.  First thanks for prompt response when I raised this.  Second, sorry I was so slow on my response to yourself.  I did go and check the availability topics you suggested.  I found them easily as per your instructions.  Of my 6 room assistant instances, only 1 was reporting offline roughly every 3.5 minutes.  It had also managed to have itself elected as leader during that period, so was taking out everything via itself.  I forced, through weights, to have another instance take over leadership.  this solved the cluster wide issue, but not the individual instance issue.  I still had some BLE tags that were truly away still attached to the instance going offline.  These BLE tags continued to have intermittent unavailable issues from the offending instance.

I re-imaged the SD card and re-installed RA on the offending PI.  Problem completely went away with a re-install.

I suspect, but am not sure, I had messed up the software updating of that one PI at some point and this somehow caused it.

Your suggestions on how to solve it sent me down the correct path and looking at correct MQTT messages.  You, for sure, helped me get it solved !

Best;

Brendan
